src/l3.c

```diff
@@ -81,7 +101,7 @@ l3_init(const char *path)
         return -1;
     }
 
-    l3_log->idx = 0;
+    memset(l3_log, 0, sizeof(L3_LOG));
```

Comment:
Safe coding. (Not sure why it was not done before.)

Reply:
This is redundant - `mmap()` guarantees that the memory it returns will be zero initialized. But, mmap will also map on demand, so doing this will touch all the pages returned by `mmap()` and depending on how much logging our program does, they might not all be used. So I think this change is not needed. Actually, on that note, we could just remove the `l3_log->idx` line, although maybe that's a bit cute.

Apologies, I should have added a comment explaining the rationale here. Perhaps you could add one?

Reply:
Done. Re-instated the `l3_log->idx = 0;`, with a comment like so:

```
104     // Technically, this is not needed as mmap() is guaranteed to return
105     // zero-filled pages. We do this just to be clear where the idx begins.
106     l3_log->idx = 0;
```